### ISSUE TITLE
Game crashes when going to video settings, launcher crashes. - https://mclo.gs/jv23peK

### ISSUE THREAD

**Issue by Daculan** [PaintNinja/SpammyCombat#1]
Heyo Ninja, so I'm having this issue, probably not updated yet, game crashes and closes even the launcher when going to the video settings. I tried the 1.19.2 43.1.1. and 1.19.2 43.1.32 and crash on both of them, if I remove the mod, it doesnt crash, heres the log. Have a good day sr, the mod is really nice tho! Ty for the work - https://mclo.gs/jv23peK

**Comment by PaintNinja** [comment]
Thanks for your report and praise, I've now fixed this in [v1.1.2](https://www.curseforge.com/minecraft/mc-mods/spammycombat/files?sort=-game-version). :)

The issue was caused by Optifine forcefully replacing Vanilla's video settings screen and laying out the buttons in a completely different way, instead of simply redirecting to its own screen. This caused SpammyCombat to think the Vanilla video settings screen was being displayed and tried grabbing from a list of buttons that Optifine removes.

I've fixed the issue by adding special detection and handling for Optifine.
